### PR TITLE
Add OperatorHub manifests

### PR DESCRIPTION
**Pull request by stefanprodan**
This PR adds the manifests need to list the operator on OperatorHub along with scorecard tests and end-to-end tests for OLM.

Part of: #8

### CODE REVIEW

**Review by souleb — APPROVED** [review]
LGTM!

File: Makefile [hunk 1]
```diff
@@ -6,6 +6,12 @@
 # Image URL to use all building/pushing image targets
 IMG ?= ghcr.io/controlplaneio-fluxcd/flux-operator:latest
 
+# FLUX_OPERATOR_VERSION refers to the version of the operator to be tested
+# under ./config/operatorhub/flux-operator/<version> directory.
+FLUX_OPERATOR_VERSION ?= v0.1.0
+# OLM_VERSION refers to the version of the Operator Lifecycle Manager to be used.
+OLM_VERSION ?= 0.28.0
```
Comment (stefanprodan) [inline Makefile:13]: This should be installed in the project bin like all others tools down below.

File: config/operatorhub/flux-operator/scripts/opm-index.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  -v /var/lib/docker:/var/lib/docker \
+  -v /var/run/docker.sock:/var/run/docker.sock \
+  opm:latest index add \
+  --container-tool docker \
+  --bundles ghcr.io/controlplaneio-fluxcd/openshift-flux-operator-catalog:bundle-"${VERSION}" \
+  --tag ghcr.io/controlplaneio-fluxcd/openshift-flux-operator-index:${VERSION}
+
+#push index
+docker push ghcr.io/controlplaneio-fluxcd/openshift-flux-operator-index:${VERSION}
+
```
Comment (stefanprodan) [inline config/operatorhub/flux-operator/scripts/opm-index.sh:43]: Ending line missing here

File: test/olm/e2e_test.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+	opm := fmt.Sprintf("ghcr.io/controlplaneio-fluxcd/openshift-flux-operator-index:%s", version)
+
+	By("loading the flux-operator bundle image on Kind")
+	err := utils.LoadImageToKindClusterWithName(img, "/test/olm")
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+	err = utils.LoadImageToKindClusterWithName(opm, "/test/olm")
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+
+	By("installing OLM")
+	cmd := exec.Command("operator-sdk", "olm", "install", "--version", olmVersion)
```
Comment (stefanprodan) [inline test/olm/e2e_test.go:47]: This should use the binary complied in the project bin dir.

File: .github/workflows/e2e-olm.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
+name: e2e-oerator-lifecycle-manager
```
Comment (stefanprodan) [inline .github/workflows/e2e-olm.yml:1]:
```suggestion
name: e2e-olm
```

File: .github/workflows/e2e-olm.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+          version: v0.23.0
+          cluster_name: kind
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@68827325e0b33c7199eb31dd4e31fbe9023e06e3 # v3.0.0
+      - name: Setup Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@d70bba72b1f3fd22344832f00baa16ece964efeb # v3.3.0
+      - name: Run OLM tests
+        run: |
+          make test-olm FLUX_OPERATOR_VERSION="${{ github.event.inputs.version }}"
```
Comment (stefanprodan) [inline .github/workflows/e2e-olm.yml:38]: `make test-olm` should install the SDK CLI in the project bin.

**Pull request merged**